index.html

```diff
@@ -1,579 +1,998 @@
+        information that is accidentally leaked.
+      </p>
+    </section>
+
+    <section class="informative">
+      <h3>Compliance</h3>
+
+      <p>
+        Review any applicable local law when considering developing or
+        operating a decentralized identifier method.
+      </p>
+
+      <p>Consider <a href="https://gdpr.eu/">GDPR.</a></p>
+
+      <p>Consider <a href="https://oag.ca.gov/privacy/ccpa">CCPA.</a></p>
+    </section>
   </section>
 
-  <section>
-    <h1>
-Future Work
-    </h1>
+  <section class="informative">
+    <h2>Security Considerations</h2>
 
-    <section>
-      <h2>
-Upper Limits on DID Character Length
-      </h2>
+    <p>
+      Decentralized Identifiers are security primitives that are often used to
+      secure important systems. This section speaks to topics that
+      implementers might consider when thinking about the security
+      characteristics of their software systems.
+    </p>
+
+    <p>
+      Consider defense in depth, cryptographic agility and political
+      acceptability of any cryptography you rely on for did method security.
+    </p>
+
+    <p>
+      Avoid complex or slow signature formats, especially if they are poorly
+      documented, or do not have an open standard with well documented test
+      vectors.
+    </p>
+
+    <p class="advisement">
+      Avoid open source implementations that are declared a "defacto
+      standard", but lack open standard technical specification.
+    </p>
+
+    <p class="advisement">
+      Add support for legacy crypto systems such as
+      <a href="https://www.iana.org/assignments/jose/jose.xhtml">jose</a>
+      and <a href="https://www.openpgp.org/">Open PGP</a>.
+    </p>
+
+    <section class="informative">
+      <h3>Vendor Lock</h3>
+      <p>
+        Competition, direct substitutability, interoperability and mutual
+        feature support are key to reducing the barrier to adoption, and
+        confidence in your DID Method.
+      </p>
+
+      <p>
+        Avoid inventing "new features", work with others to find a common way
+        to express any new features that is not unique to your did method.
+      </p>
 
       <p>
-The current specification does not take a position on maximum length of a
-<a>DID</a>. The maximum interoperable URL length is currently about 2K
-characters. QR codes can handle about 4K characters. Clients using <a>DIDs</a>
-will be responsible for storing many <a>DIDs</a>, and some methods would be able
-to externalize some of their costs onto clients by relying on more complicated
-signature schemes or by adding state into <a>DIDs</a> intended for temporary
-use. A future version of this specification should set reasonable limits on
-<a>DID</a> character length to minimize externalities.
+        Avoid hard coupling to specific networks, such as Bitcoin or Hyper
+        Ledger Fabric. Design your method such that it may be adapted to
+        support multiple ledger systems.
```

Comment:
```suggestion
        Avoid hard coupling to specific networks, such as Bitcoin or
        Hyperledger Fabric. Design your method such that it may be adapted
        to support multiple ledger systems.
```